src/model/dla.py

```diff
@@ -0,0 +1,390 @@
+        assert isinstance(padding, int), 'The padding whould be given as an int'
+
+        self.custom_padding = padding
+        self.custom_padding_tensor = tf.constant([[0, 0], [0, 0], [padding, padding], [padding, padding]])
+        if data_format == 'channels_first':
+            self.custom_padding_tensor = tf.constant([[0, 0], [0, 0], [padding, padding], [padding, padding]])
+        else:
+            self.custom_padding_tensor = tf.constant([[0, 0], [padding, padding], [padding, padding], [0, 0]])
+
+        super().__init__(filters=filters,
+                         kernel_size=kernel_size,
+                         strides=strides,
+                         padding='valid',
+                         data_format=data_format,
+                         dilation_rate=dilation_rate,
+                         groups=groups,
+                         activation=activation,
+                         use_bias=use_bias,
+                         kernel_initializer=kernel_initializer,
+                         bias_initializer=bias_initializer,
+                         kernel_regularizer=kernel_regularizer,
+                         bias_regularizer=bias_regularizer,
+                         activity_regularizer=activity_regularizer,
+                         kernel_constraint=kernel_constraint,
+                         bias_constraint=bias_constraint,
+                         **kwargs)
+
+    def build(self, input_shape):
+        self.orig_input_shape = input_shape
+        super().build(tf.TensorShape(np.array(input_shape) + self.custom_padding_tensor.numpy().sum(axis=1)))
+
+    def call(self, inputs):
+        x = tf.pad(inputs, self.custom_padding_tensor)
+        x = super().call(x)
+        return x
+
+
+class BasicBlock(tf.keras.Model):
+    def __init__(self, planes, stride=1, dilation=1):
+        super(BasicBlock, self).__init__()
+        self.conv1 = CustomConv2d(planes, kernel_size=3,
+                                  strides=stride, padding=dilation,
+                                  use_bias=False, dilation_rate=dilation, data_format='channels_first')
+        self.bn1 = BatchNorm2d(momentum=BN_MOMENTUM)
+        self.relu = layers.ReLU()
+        self.conv2 = CustomConv2d(planes, kernel_size=3,
+                                  strides=1, padding=dilation,
+                                  use_bias=False, dilation_rate=dilation, data_format='channels_first')
+        self.bn2 = BatchNorm2d(momentum=BN_MOMENTUM)
+        self.stride = stride
+
+    def call(self, x, residual=None):
+        if residual is None:
+            residual = x
+
+        out = self.conv1(x)
+        out = self.bn1(out)
+        out = self.relu(out)
+
+        out = self.conv2(out)
+        out = self.bn2(out)
+
+        out += residual
+        out = self.relu(out)
+
+        return out
+
+
+class Root(tf.keras.Model):
+    def __init__(self, out_channels, kernel_size, residual):
+        super(Root, self).__init__()
```

Comment:
Just for consistency with the other `super()` calls, remove the arguments here.

Reply:
also fixed the rest of `super()` calls